File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int biggest = nums[0];
        int total = 0;

        int i = 0;
        while(i + 1 < nums.size() && nums[i + 1] == nums[i] + 1){
            biggest += nums[i + 1];
            i++;
        }
        total = biggest;

        unordered_set<int> st(nums.begin(), nums.end());

        while(st.count(total)){
            total++;
        }
        return total;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        total = nums[0]
        i = 0

        while i + 1 < len(nums) and nums[i + 1] == nums[i] + 1:
            i += 1
            total += nums[i]

        biggest = total

        while biggest in nums:
            biggest += 1

        return biggest